spring-integration-core/src/main/java/org/springframework/integration/gateway/RequestReplyExchanger.java

```diff
@@ -26,6 +26,7 @@
  * @author Mark Fisher
  * @since 2.0
  */
+@FunctionalInterface
 public interface RequestReplyExchanger {
 
 	Message<?> exchange(Message<?> request);
```

Comment:
We talked about `throws MessagingException` here - let's not forget - either do it here or create a JIRA.